File: src/builder/examples/petreport/PetReport.java
package builder.examples.petreport;

import petapp.domain.Location;
import petapp.domain.Owner;
import petapp.domain.Pet;

public class PetReport {

    private Pet pet;
    private Owner owner;
    private Location location;
    private long reportedAtEpochMs;

    public Pet getPet() { return pet; }
    public void setPet(Pet pet) { this.pet = pet; }

    public Owner getOwner() { return owner; }
    public void setOwner(Owner owner) { this.owner = owner; }

    public Location getLocation() { return location; }
    public void setLocation(Location location) { this.location = location; }

    public long getReportedAtEpochMs() { return reportedAtEpochMs; }
    public void setReportedAtEpochMs(long reportedAtEpochMs) { this.reportedAtEpochMs = reportedAtEpochMs; }

    @Override
    public String toString() {
        return String.format("PetReport{%s, at=%s, by=%s}",
                pet, location, owner != null ? owner.getFullName() : "unknown");
    }
}

File: src/builder/examples/petreport/PetReport.js
'use strict';

class PetReport {
  constructor() {
    this.pet = null;
    this.owner = null;
    this.location = null;
    this.reportedAtEpochMs = 0;
  }

  toString() {
    const at = this.location ? this.location.toString() : '?';
    const by = this.owner ? this.owner.fullName : 'unknown';
    return `PetReport{${this.pet}, at=${at}, by=${by}}`;
  }
}

module.exports = PetReport;
